File: services/internal-treasury/src/main/java/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/mappers/IdempotencyKeyMapper.java
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.mappers;

import br.com.deltaglobalbank.internal_treasury.domain.idempotency.IdempotencyKey;
import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.IdempotencyKeyEntity;

public final class IdempotencyKeyMapper {

    private IdempotencyKeyMapper() {
    }

    public static IdempotencyKey toDomain(IdempotencyKeyEntity entity) {
        return new IdempotencyKey(
            entity.getKey(),
            entity.getResponse(),
            entity.getCreatedAt()
        );
    }

    public static IdempotencyKeyEntity toEntity(IdempotencyKey domain) {
        return new IdempotencyKeyEntity(
            domain.key(),
            domain.response(),
            domain.createdAt()
        );
    }
}

File: services/internal-treasury/src/main/kotlin/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/mappers/IdempotencyKeyMapper.kt
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.mappers

import br.com.deltaglobalbank.internal_treasury.domain.idempotency.IdempotencyKey
import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.IdempotencyKeyEntity

fun IdempotencyKeyEntity.toDomain(): IdempotencyKey = IdempotencyKey(
    key = this.key,
    response = this.response,
    createdAt = this.createdAt
)

fun IdempotencyKey.toEntity(): IdempotencyKeyEntity = IdempotencyKeyEntity(
    key = this.key,
    response = this.response,
    createdAt = this.createdAt,
)